### ISSUE TITLE
Hello, I would like to ask about the installation of pointnet2_batch.

### ISSUE THREAD

**Issue by utilpotatocat** [mlvlab/SPoTr#15]
Does this require the cuda version of the system to reach 11.3 instead of the cuda-kit in the virtual environment reaching 11.3?
Because during installation, Python's setup function will automatically read the cuda version in your system by default instead of the virtual environment you set. To be honest, this is very unfriendly to multi-user server users.
Do you have any solutions?
![image](https://github.com/mlvlab/SPoTr/assets/108716614/65cea450-03f5-40e1-914f-cd4f798cf2df)
As shown in the picture above, the administrator of the server I use set the system cuda version to 12.0. So even if I install the 11.3 cudatoolkit in my conda virtual environment, I will also encounter such an error message. Because the setup function in python's setuptools library will point to the system's cuda version by default.
![image](https://github.com/mlvlab/SPoTr/assets/108716614/81c25669-36c6-4d3c-a35f-4204aa23cfa3)

**Comment by utilpotatocat** [comment]
I seem to have found the _check_cuda_version function in the setup function in the setuptools library on the Internet.

```
def _check_cuda_version(self):
        if CUDA_HOME:
            nvcc = os.path.join(CUDA_HOME, 'bin', 'nvcc')
            cuda_version_str = subprocess.check_output([nvcc, '--version']).strip().decode(*SUBPROCESS_DECODE_ARGS)
            cuda_version = re.search(r'release (\d+[.]\d+)', cuda_version_str)
            if cuda_version is not None:
                cuda_str_version = cuda_version.group(1)
                cuda_ver = packaging.version.parse(cuda_str_version)
                torch_cuda_version = packaging.version.parse(torch.version.cuda)
                if cuda_ver != torch_cuda_version:
                    # major/minor attributes are only available in setuptools>=49.6.0
                    if getattr(cuda_ver, "major", float("nan")) != getattr(torch_cuda_version, "major", float("nan")):
                        raise RuntimeError(CUDA_MISMATCH_MESSAGE.format(cuda_str_version, torch.version.cuda))
                    warnings.warn(CUDA_MISMATCH_WARN.format(cuda_str_version, torch.version.cuda))

        else:
            raise RuntimeError(CUDA_NOT_FOUND_MESSAGE)
```

This seems to indicate that the setup function that comes with python points to the cuda version of your system by default (in your CUDA_HOME).
This seems to indicate that this project must reinstall cuda at the system level.
This is very troublesome, especially for those multi-user server users. Do you have any better solutions? Is there any possibility of not reinstalling CUDA at the system level?

**Comment by PJin0** [comment]
I appreciate your interest in our paper.

I haven't encountered this kind of error, so I have no solutions for it. Sorry about that.

Our codes are based on OpenPoints library (https://github.com/guochengqian/openpoints). So, I recommend you to discuss this issue on the repo of the OpenPoints library. Thank you:)
